### PR TITLE
test: split test setup into a separate library

### PR DESCRIPTION
**Pull request by Tlacenka**
In this PR, I split testing utilities into two libraries - `test-utils` and `test-setup`.
Thanks to this, any integrations introduced in `utils` can be imported in `test-setup` as it no longer introduces a circular dependency.

## Nx graph (without the logger dependency)
image

### CODE REVIEW

File: testing/test-setup/src/lib/git.mock.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { vi } from 'vitest';
-import { MEMFS_VOLUME } from '../constants';
+import { MEMFS_VOLUME } from '../../../test-utils/src/lib/constants';
```
Comment (matejchalk) [inline testing/test-setup/src/lib/git.mock.ts:2]:
```suggestion
import { MEMFS_VOLUME } from '@code-pushup/test-utils';
```

**Review by matejchalk — APPROVED** [review]

**Comment by BioPhoton** [comment]
I'm a bit confused... we did discuss this in slack, I sayed ill do it, did even notify that im half done and get food and now there is a new PR... Did I missed something?

**Review by BioPhoton — COMMENTED** [review]
Nice!

**Review by BioPhoton — APPROVED** [review]

File: testing/test-setup/src/lib/git.mock.ts (hunk 1 above)
Comment (Tlacenka) [reply testing/test-setup/src/lib/git.mock.ts]: Nice catch, fixed.

**Review by matejchalk — APPROVED** [review]

**Review by BioPhoton — APPROVED** [review]
thx for the nice graphic! Looking forward to merge this

**Pull request merged**